modelstore/models/causalml.py

```diff
@@ -0,0 +1,71 @@
+#    Copyright 2024 Neal Lathia
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+import os
+from functools import partial
+from typing import Any
+
+from modelstore.metadata import metadata
+from modelstore.models.common import load_joblib, save_joblib
+from modelstore.models.model_manager import ModelManager
+from modelstore.storage.storage import CloudStorage
+
+MODEL_FILE = "model.joblib"
+
+
+class CausalMLManager(ModelManager):
+
+    """
+    Model persistence for Causal ML models:
+    https://causalml.readthedocs.io/en/latest/index.html
+    """
+
+    NAME = "causalml"
+
+    def __init__(self, storage: CloudStorage = None):
+        super().__init__(self.NAME, storage)
+
+    def required_dependencies(self) -> list:
+        return ["causalml"]
+
+    def optional_dependencies(self) -> list:
+        deps = super().optional_dependencies()
+        return deps + ["causalml"]
```

Comment:
fyi, since `causalml` is a required dependency, then you shouldn't need to list it as optional as well

Reply:
Ah that makes sense

Are optional dependencies purely for logging purposes? Wasn't sure how many to add here 😅

Reply:
yeah, that's right! You can leave this out altogether if you want ([here's the default set](https://github.com/operatorai/modelstore/blob/main/modelstore/models/model_manager.py#L53-L56)) and then these just get saved in the model's meta data [here](https://github.com/operatorai/modelstore/blob/main/modelstore/models/model_manager.py#L216)